### DISCUSSION TITLE
problem with building v3.6.0 in shell_stop

### DISCUSSION THREAD

**Discussion by lukas-fwdev** [zephyrproject-rtos/zephyr#69553] [Q&A] [answered]
Hi,
I am facing a problem when migrating from v3.5.0 to v3.6.0.
I get the error:
`c:/myPath/zephyr-sdk-0.16.1/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.2.0/../../../../arm-zephyr-eabi/bin/ld.bfd.exe: zephyr/libzephyr.a(shell.c.obj): in function 'shell_stop':
P:/myPath/zephyr/subsys/shell/shell.c:1463: undefined reference to 'z_shell_log_backend_disable'
collect2.exe: error: ld returned 1 exit status`

I can see that the call of z_shell_log_backend_disable() inside shell_stop function was introduced with PR #67400 from @jakub-uC . 
I am wondering if there is missing a guard around this call like it is done in function instance_uninit() in the same file.
When changing the simple function call to:
`if (IS_ENABLED(CONFIG_SHELL_LOG_BACKEND)) {
		z_shell_log_backend_disable(sh->log_backend);
	}`
everything is working fine.

Does anyone of you know, if I just missed any configuration or similiar, or if this is a real bug?

Thanks in advance!

**Comment by GeorgeCGV** [comment, marked as answer]
The `z_shell_log_backend_disable` won't be available if `CONFIG_SHELL_LOG_BACKEND` is off.
It is a bug, forgotten `IS_ENABLED(CONFIG_SHELL_LOG_BACKEND)` condition.

**Reply by lukas-fwdev** [reply]
Thanks, I opened this as an issue.
